server_run.py

```diff
@@ -30,6 +31,17 @@ def server_run(args):
     set_scoped_session(scoped_session(sessionmaker(bind=engine),
                                       scopefunc=lambda: _request_ctx_stack.top))
 
+    if args.recreate_db:
+        if args.debug:
+            exit("Not recreating the with `--debug`, "
+                 "which usually leads to multiple restarts on file change.\n"
+                 "This would override any database change every time.")
+        print("Dropping pycroft model schema")
+        drop_db_model(engine)
```

Comment:
This won't work if tables are renamed or removed.

The new model won't contain the old tables/indexes/etc. and therefore can't drop them.